### PR TITLE
fix(Overseerr): use rewrites as a substitute for BaseURL

### PR DESCRIPTION
**Pull request by brettpetch**
## Description
Devs released a working nginx config for this.

## Fixes issues: 
- Un-tracked issue...
- rel: https://github.com/swizzin/swizzin/pull/567

## Proposed Changes:
- Apply fix to nginx config to allow for overseer to be put into a folder.

## Change Categories
- Bug fix 

## Checklist
- [x] Docs have been made OR are not necessary
    - PR link: 
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas

### Architectures

|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|	⚫️		|				|
| Bionic		|			|			|			|				|
| Buster		|	⚫️		|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|			|			|			|				|

### 🛠🛠 TODO
- [x] Shellcheck is yelling, find out why / create ignore

### ❌❌ Currently failing
Shell check is yelling at me and I don't know why!

Passed testing tho...

### CODE REVIEW

**Comment by brettpetch** [comment]
@flying-sausages <3

File: scripts/nginx/overseerr.sh [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+    sub_filter '/api/v1' '/\$app/api/v1';
+    sub_filter '/login/plex/loading' '/\$app/login/plex/loading';
+    sub_filter '/images/' '/\$app/images/';
+    sub_filter '/android-' '/\$app/android-';
+    sub_filter '/apple-' '/\$app/apple-';
+    sub_filter '/favicon' '/\$app/favicon';
+    sub_filter '/logo.png' '/\$app/logo.png';
+    sub_filter '/site.webmanifest' '/\$app/site.webmanifest';
 }
 EOF
```
Comment (brettpetch) [inline scripts/nginx/overseerr.sh:30]: shell check is yelling about this; not entirely sure why.

**Review by brettpetch — COMMENTED** [review]
shellcheck issue stems from here. anyone got an idea?

File: scripts/nginx/overseerr.sh (hunk 1 above)
Comment (flying-sausages) [reply scripts/nginx/overseerr.sh:30]: I think the shellcheck is failing because the container is failing to build, nothing with your code.

**Comment by flying-sausages** [comment]
I hope this is not the way this will get implemented lmao I really don't want to support this.

For that reason, I'm lowkey hesitant about including this. If we release this to master then we'll have to patch it out when they figure it out...

And then imagine this thing breaks on its own when they make changes and we have to forcefully make everyone upgrade their instances and shit yuck.

I'll think about this. It might just be the only option we'll ever get, really...

**Comment by brettpetch** [comment]
> I hope this is not the way this will get implemented lmao I really don't want to support this.
> 
> For that reason, I'm lowkey hesitant about including this. If we release this to master then we'll have to patch it out when they figure it out...
> 
> And then imagine this thing breaks on its own when they make changes and we have to forcefully make everyone upgrade their instances and shit yuck.
> 
> I'll think about this. It might just be the only option we'll ever get, really...

Yeah, it does seem far from ideal. Seems really hacky; but was sort-of a solution for the time being from what I read in the docs.

**Comment by flying-sausages** [comment]
I tried this out and it is way too hacky for anything usable. It really _barely_ works, you click on the wrong thing in the UI and it's broken. Refresh outside the homepage and it's broken. I'm hesitant to merge this even into the beta branch.

**Comment by brettpetch** [comment]
> I tried this out and it is way too hacky for anything usable. It really barely works, you click on the wrong thing in the UI and it's broken. Refresh outside the homepage and it's broken. I'm hesitant to merge this even into the beta branch.

Yeah, it really isn't the ideal implementation. Lets hope devs get around to adding proper baseurl support

/b

**Comment by flying-sausages** [comment]
If you may, please keep the branch so that others can use it if they wanna

**Comment by brettpetch** [comment]
Reopening for folks may use the branch.

**Comment by flying-sausages** [comment]
I'm gonna contradict myself there and re-close it. This PR is linked to from the other one, and as long as the original branch on your fork does not get deleted then it will remain accessible. We can always link it to people in the discord too.

**Pull request closed without merging**